api/pkg/transformer/feast/enricher.go

```diff
@@ -0,0 +1,73 @@
+package feast
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/buger/jsonparser"
+	"github.com/opentracing/opentracing-go"
+	"go.uber.org/zap"
+
+	"github.com/gojek/merlin/pkg/transformer"
+	"github.com/gojek/merlin/pkg/transformer/types"
+)
+
+// Enricher wraps feast serving client to retrieve features.
+type Enricher struct {
+	featureRetriever FeatureRetriever
+	logger           *zap.Logger
+}
+
+// NewEnricher initializes a new Enricher.
+func NewEnricher(featureRetriever FeatureRetriever, logger *zap.Logger) (*Enricher, error) {
+	return &Enricher{
+		featureRetriever: featureRetriever,
+		logger:           logger,
+	}, nil
+}
+
+// Enrich retrieves the Feast features values and add them into the request.
+func (t *Enricher) Enrich(ctx context.Context, request []byte, _ map[string]string) ([]byte, error) {
```

Comment:
why do you need `_map[string]string`?

Reply:
This is to pass request headers, it's not used in feast enricher.

Reply:
why pass this if it is not being used? is it part of the interface method?

Reply:
Yes, it's part of interface